Sitka-Sound-Science-Center/sitka-landslide
# Project development setup

## Overview

Sets up some of the standard project infrastructure:
- Scripts To Rule Them All: Just dev-related scripts so far (setup, update, server, lint, and format)
- Prettier:
   - Added to frontend dev dependencies
   - Project-level `.prettierrc` file
   - Runs in "check" mode in `lint` and "write" mode in `format`
   - I ran it and committed the changes it made
- README: The frontend had a standard Next.js README already, which I revised a bit but didn't delete. I added a top-level README with the standard setup instructions and list of scripts and ports.
- Dev server port: Picked a port (3008), recorded it in [the team Ports list](https://github.com/azavea/geospatial-apps/blob/master/project_ports.md), and changed all references to the default one (3000).
- Pull request template

Resolves #1

### Demo

![image](https://user-images.githubusercontent.com/6598836/155161518-31f30ff3-6fc6-45d8-9b7d-023e69072e31.png)

### Notes

It's possible we won't benefit in the long term from having the front-end in a subdirectory, in which case we might decide that setting it up that way and having to deal with that nuance (e.g. have all the STRTA change directory before they do anything) is mildly annoying on balance.  But if we do end up with a few different components here, which seems likely given that we're planning to make the data processing independent from the front-end, I think it'll be nice to not have things mixed together.

## Testing Instructions

- Run `./scripts/setup` (which actually just calls `update`)
- Run `./scripts/update` (it won't do much, since you just ran it)
- Run `./scripts/server`. Go to http://localhost:3008/ and you should see a sample graph and a links to a few other preliminary snippets.
- Run `./scripts/lint`. It should run successfully and not find any errors.
- Peruse the READMEs for missing or erroneous info, or typos.

## Checklist
